src/Type/Union/PostObjectUnion.php

```diff
@@ -5,7 +5,7 @@
 use WPGraphQL\Types;
 
 $possible_types     = [];
-$allowed_post_types = \WPGraphQL::get_allowed_post_types();
+$allowed_post_types = get_post_types( [ 'show_in_graphql' => true ] );
```

Comment:
@kidunot89 wait, why is this changing? Don't we want your filter to be applied here too?

Like if a user were to do a query across many post_types, WordPress might return results of posts, pages and Products, and this union should know that the Product post_type should be included in the union, which your filter applied to WPGraphQL::get_allowed_post_types() would include. 

Can you elaborate on why this change is necessary?

I would argue that it's not

Reply:
If `\WPGraphQL::get_allowed_post_types()` is the Product post-type won't be seen due the hook in the example in the description. If `get_post_types( [ 'show_in_graphql' => true ] )` is the hook is bypassed, and the `Product` post-type is included. Which is the desired behavior.